src/main/java/com/raeic/embarker/party/models/PartyPlayer.java

```diff
@@ -20,6 +20,10 @@ public String getLeaderUniqueId() {
     }
 
     public void delete() {
+        // Invalidate the cache on the party leader's Party object, as the
+        // party's been updated.
+        Globals.party.invalidateCacheByKey(leaderUniqueId);
```

Comment:
There's a race condition here, we need to invalidate the cache after it's deleted from SQL